src/main/java/org/folio/circulation/domain/AddressType.java

```diff
@@ -0,0 +1,40 @@
+package org.folio.circulation.domain;
+
+import static org.folio.circulation.support.JsonPropertyFetcher.getProperty;
+
+import io.vertx.core.json.JsonObject;
+
+
+public class AddressType {
+  private final String addressType;
+  /**
+   * Address Type description
+   */
+  private final String desc;
+  private final String id;
+
+
+  public AddressType(JsonObject representation) {
```

Comment:
getProperty method already checks for nulls. Also because fields are final if I add null check, I would have to explicitly set all fields to null like this:
```
if(representation !=null){
    this.addressType = getProperty(representation, "addressType");
    this.description = getProperty(representation, "desc");
    this.id = getProperty(representation, "id");
}
else{
    this.addressType = null;
    this.description = null;
    this.id = null;
}
```
so it seems better to follow the pattern used in other model classes and rely on getProperty

Reply:
ok, as long as we don't run into NullPointerException somewhere down the line.